### PR TITLE
Unify ConfigMap sync function

### PR DESCRIPTION
**Pull request by sjug**
Just a bit of cleanup, if there are ConfigMap update issues I will address here too.

### CODE REVIEW

**Comment by sjug** [comment]
Things look pretty happy after moving to the single namespace. Secondary watch on the DaemonSet works as well, updates work.

File: assets/tuned/03-cluster-role-binding.yaml [hunk 1]
```diff
@@ -5,10 +5,9 @@ metadata:
...
   kind: ClusterRole
   name: cluster-node-tuning:tuned
-  namespace: openshift-cluster-node-tuning
 subjects:
 - kind: ServiceAccount
   name: tuned
-  namespace: openshift-cluster-node-tuning
+  namespace: openshift-cluster-node-tuning-operator
 userNames:
 - system:serviceaccount:openshift-cluster-node-tuning:tuned
```
Comment (jmencak) [inline assets/tuned/03-cluster-role-binding.yaml]: `system:serviceaccount:openshift-cluster-node-tuning:tuned` -> `system:serviceaccount:openshift-cluster-node-tuning-operator:tuned`, please

Comment (sjug) [reply assets/tuned/03-cluster-role-binding.yaml]: Done

**Comment by jmencak** [comment]
I was not able to successfully test on 4.0 using an image until the following changes.

```
diff --git a/manifests/04-rbac.yaml b/manifests/04-rbac.yaml
index 2a2f9db..06e6a3b 100644
--- a/manifests/04-rbac.yaml
+++ b/manifests/04-rbac.yaml
@@ -19,17 +19,17 @@ rules:
   verbs: ["list","watch"]
 - apiGroups: ["apps"]
   resources: ["daemonsets"]
-  verbs: ["create","delete","update"]
+  verbs: ["create","delete","list","update","watch"]
 - apiGroups: ["rbac.authorization.k8s.io"]
   resources: ["clusterroles","clusterrolebindings"]
-  verbs: ["create","list","watch"]
+  verbs: ["create","list","watch","update"]
 - apiGroups: ["security.openshift.io"]
   resources: ["securitycontextconstraints"]
   verbs: ["use"]
 # "" indicates the core API group
 - apiGroups: [""]
   resources: ["configmaps","namespaces","serviceaccounts","services"]
-  verbs: ["create","delete","update"]
+  verbs: ["create","get","delete","list","update","watch"]
 - apiGroups: [""]
   resources: ["nodes"]
   verbs: ["get","list","watch"]
diff --git a/pkg/controller/tuned/tuned_controller.go b/pkg/controller/tuned/tuned_controller.go
index 18796b5..d596eb0 100644
--- a/pkg/controller/tuned/tuned_controller.go
+++ b/pkg/controller/tuned/tuned_controller.go
@@ -87,7 +87,6 @@ func syncServiceAccount(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 	if err != nil {
 		return fmt.Errorf("Couldn't build tuned ServiceAccount: %v", err)
 	}
-	controllerutil.SetControllerReference(tuned, saManifest, r.scheme)
 	log.Printf("Built ServiceAccount manifest for %s/%s\n", saManifest.Namespace, saManifest.Name)
 
 	sa := &corev1.ServiceAccount{}
@@ -98,6 +97,7 @@ func syncServiceAccount(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 			return fmt.Errorf("Couldn't create tuned ServiceAccount: %v", err)
 		}
 		log.Printf("Created ServiceAccount for %s/%s\n", saManifest.Namespace, saManifest.Name)
+		controllerutil.SetControllerReference(tuned, saManifest, r.scheme)
 	} else if err != nil {
 		log.Printf("Failed to get ServiceAccount: %v\n", err)
 		return err
@@ -119,7 +119,6 @@ func syncClusterRole(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 		return fmt.Errorf("Couldn't build tuned ClusterRole: %v", err)
 	}
 	log.Printf("Built ClusterRole manifest for %s\n", crManifest.Name)
-	controllerutil.SetControllerReference(tuned, crManifest, r.scheme)
 
 	cr := &rbacv1.ClusterRole{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: crManifest.Name, Namespace: ""}, cr)
@@ -129,6 +128,7 @@ func syncClusterRole(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 			return fmt.Errorf("Couldn't create tuned ClusterRole: %v", err)
 		}
 		log.Printf("Created ClusterRole for %s\n", crManifest.Name)
+		controllerutil.SetControllerReference(tuned, crManifest, r.scheme)
 	} else if err != nil {
 		log.Printf("Failed to get ClusterRole: %v\n", err)
 		return err
@@ -150,7 +150,6 @@ func syncClusterRoleBinding(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error
 		return fmt.Errorf("Couldn't build tuned ClusterRoleBinding: %v", err)
 	}
 	log.Printf("Built ClusteRoleBinding manifest for %s\n", crbManifest.Name)
-	controllerutil.SetControllerReference(tuned, crbManifest, r.scheme)
 
 	crb := &rbacv1.ClusterRoleBinding{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: crbManifest.Name, Namespace: ""}, crb)
@@ -160,6 +159,7 @@ func syncClusterRoleBinding(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error
 			return fmt.Errorf("Couldn't create tuned ClusterRoleBinding: %v", err)
 		}
 		log.Printf("Created ClusterRoleBinding for %s\n", crbManifest.Name)
+		controllerutil.SetControllerReference(tuned, crbManifest, r.scheme)
 	} else if err != nil {
 		log.Printf("Failed to get ClusterRoleBinding: %v\n", err)
 		return err
@@ -181,7 +181,6 @@ func syncClusterConfigMap(f func(tuned *tunedv1alpha1.Tuned) (*corev1.ConfigMap,
 		return fmt.Errorf("Couldn't build tuned ConfigMap: %v", err)
 	}
 	log.Printf("Built ConfigMap manifest for %s/%s\n", cmManifest.Namespace, cmManifest.Name)
-	controllerutil.SetControllerReference(tuned, cmManifest, r.scheme)
 
 	cm := &corev1.ConfigMap{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: cmManifest.Name, Namespace: cmManifest.Namespace}, cm)
@@ -191,6 +190,7 @@ func syncClusterConfigMap(f func(tuned *tunedv1alpha1.Tuned) (*corev1.ConfigMap,
 			return fmt.Errorf("Couldn't create tuned ConfigMap: %v", err)
 		}
 		log.Printf("Created ConfigMap for %s/%s\n", cmManifest.Namespace, cmManifest.Name)
+		controllerutil.SetControllerReference(tuned, cmManifest, r.scheme)
 	} else if err != nil {
 		log.Printf("Failed to get ConfigMap: %v\n", err)
 		return err
@@ -216,7 +216,6 @@ func syncDaemonSet(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 		return fmt.Errorf("Couldn't build tuned DaemonSet: %v", err)
 	}
 	log.Printf("Built DaemonSet manifest for %s/%s\n", dsManifest.Namespace, dsManifest.Name)
-	controllerutil.SetControllerReference(tuned, dsManifest, r.scheme)
 
 	daemonset := &appsv1.DaemonSet{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: dsManifest.Name, Namespace: dsManifest.Namespace}, daemonset)
@@ -226,6 +225,7 @@ func syncDaemonSet(r *ReconcileTuned, tuned *tunedv1alpha1.Tuned) error {
 			return fmt.Errorf("Couldn't create tuned DaemonSet: %v", err)
 		}
 		log.Printf("Created DaemonSet for %s/%s\n", dsManifest.Namespace, dsManifest.Name)
+		controllerutil.SetControllerReference(tuned, dsManifest, r.scheme)
 	} else if err != nil {
 		log.Printf("Failed to get DaemonSet: %v\n", err)
 		return err
```

Please note I haven't tested the controller references are actually being set and they work as they should.  If you insist on having controllerutil.SetControllerReference() in this PR, please check the return status and act upon it.  I'd rather keep this PR small and make sure it fixes issues introduced by PR14 to make the cluster-node-tuning-operator works out-of-the-box again.  Thanks.

**Comment by sjug** [comment]
@jmencak Do you think we should create a bugzilla against 3.11/3.x RBAC given it doesn't seem to follow the same rules?

Setting the owner reference was causing breakage in 4.0 testing?

**Comment by jmencak** [comment]
I haven't tested against 3.10/3.11, so cannot comment.  As for pulling the owner reference, @sjug , I'm inclined to say yes, please pull them and let's work on it in a separate PR.

**Comment by sjug** [comment]
@jmencak As discussed, pulled `controllerReference` and squashed for final testing and sign-off.

**Pull request merged**